### DISCUSSION TITLE
Should `description` be added into `APIRouter `

### DISCUSSION THREAD

**Discussion by monchin** [fastapi/fastapi#11833] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
There is no code need in this discussion.
```

### Description

Hi, I'm trying to use `APIRouter` as my `FastAPI` app becomes big and I need to classify different functions to different routers. But I found that where `FastAPI` has `description` arg, which can be easily read in `/docs`, while [`APIRouter`](https://fastapi.tiangolo.com/reference/apirouter/#fastapi.APIRouter) does not have this arg. In my opinion, a description to the whole router would sometimes be helpful, although we can add description to every function. So maybe it's a good idea. I don't know if this is because `openapi` doesn't support or just because we didn't implement it?

### Operating System

Linux

### FastAPI Version

0.111.0

### Pydantic Version

2.8.0

### Python Version

3.10.12

**Comment by PhysicallyActive** [comment, marked as answer]
Why not add one or multiple `tags` to your routers and then outlining what the different tags do in your application description?

```python
import fastapi
from fastapi import APIRouter, FastAPI

router = APIRouter(tags=["my-tag"])


@router.get("/hello")
async def hello_world() -> str:
    return "Hello, World!"


app = FastAPI(
    description="""
My API.

## Tags

### my-tag
You can do stuff here.
""",
)
app.include_router(router)
```

------

Personally, I don't see the gain in adding a `description` field to each router since the routers aren't supposed to represent unique "sub-applications", but rather a way for you (the programmer) to categorise your application paths together in a more structured way.

**Reply by monchin** [reply]
Thank you for your great method!
